src/ancillary.rs

```diff
@@ -287,11 +286,11 @@ impl AsMut<[u8]> for AncillaryBuf {
 
 
 /// One ancillary message produced by [`Ancillary`](#struct.Ancillary)
-pub enum AncillaryItem<'a> {
+pub enum AncillaryItem {
     /// One or more file descriptors sent by the peer.
     ///
     /// Consumer of the iterator is responsible for closing them.
-    Fds(&'a[RawFd]),
+    Fds(Vec<RawFd>),
```

Comment:
Allocating just for one or two fds would be unfortunate.
I'd rather add a new struct that permits the unalignedness.

Something like:
```rust
struct FdSlice<'a> {
    unaligned_ptr: *const RawFd,
    len: usize,
    _borrow: PhantomData<&RawFd>,
}
```
but I can do that.

Reply:
Small allocations should be somewhat negligible (compared to the overhead of syscalls and unix sockets) but `FdSlice` sounds like a neat solution

Reply:
Removed Vec handling for now.